snmplib/transports/snmpUDPDomain.c

```diff
@@ -98,6 +98,58 @@ netsnmp_udp_fmtaddr(netsnmp_transport *t, const void *data, int len)
     return netsnmp_ipv4_fmtaddr("UDP", t, data, len);
 }
 
+static int
```

Comment:
Please split this change into two patches: one patch that introduces the netsnmp_udp_resolve_source() function and a second patch that adds the '@' netgroup functionality.